app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -76,6 +77,54 @@ class OnboardingFragmentPresenter @Inject constructor(
       })
   }
 
+  private fun createRecyclerViewAdapter(): BindableAdapter<OnBoardingViewPagerViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<OnBoardingViewPagerViewModel, ViewType> { viewModel ->
+        when (viewModel) {
+          is OnboardingSlideViewModel -> ViewType.ONBOARDING_SLIDE
+          is OnboardingSlideFinalViewModel -> ViewType.ONBOARDING_FINAL_SLIDE
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+        }
+      }
+      .setLifecycleOwner(fragment)
+      .registerViewBinder(
+        viewType = ViewType.ONBOARDING_SLIDE,
+        inflateView = { parent ->
+          OnboardingSlideBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            /* attachToParent= */ false
+          ).root
+        },
+        bindView = { view, viewModel ->
+          val binding = DataBindingUtil.findBinding<OnboardingSlideBinding>(view)!!
+          val onboardingSlideViewModel = viewModel as OnboardingSlideViewModel
+          binding.viewModel = onboardingSlideViewModel
+        }
+      )
+      .registerViewBinder(
+        viewType = ViewType.ONBOARDING_FINAL_SLIDE,
+        inflateView = { parent ->
+          OnboardingSlideFinalBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            /* attachToParent= */ false
+          ).root
+        },
+        bindView = { view, viewModel ->
+          val binding = DataBindingUtil.findBinding<OnboardingSlideFinalBinding>(view)!!
+          val onboardingSlideFinalViewModel = viewModel as OnboardingSlideFinalViewModel
+          binding.viewModel = onboardingSlideFinalViewModel
+        }
+      )
+      .build()
+  }
+
+  private enum class ViewType {
+    ONBOARDING_SLIDE,
```

Comment:
Perhaps instead: ONBOARDING_MIDDLE_SLIDE vs. FINAL? The two types don't currently seem mutually exclusive in concept, but as distinct view types (& enum constants) they should be.

Reply:
Done.